vm/systemSmartContracts/staking.go

```diff
@@ -499,15 +508,26 @@ func (r *stakingSC) processStake(blsKey []byte, registrationData *StakedDataV2_0
 		return nil
 	}
 
-	r.addToStakedNodes()
-	registrationData.Staked = true
-	registrationData.StakedNonce = r.eei.BlockChainHook().CurrentNonce()
-	registrationData.UnStakedEpoch = core.DefaultUnstakedEpoch
-	registrationData.UnStakedNonce = 0
+	err := r.removeFromWaitingList(blsKey)
+	if err != nil {
+		r.eei.AddReturnMessage("error while removing from waiting")
+		return err
+	}
+	r.addToStakedNodes(1)
+	r.activeStakingFor(registrationData)
```

Comment:
Do you need to r.eei.Finish([]byte{waiting/staking}) before return here?

Reply:
Return finish must be added for waiting key only.